src/HttpClientFactory/Http/ref/Microsoft.Extensions.Http.netcoreapp.cs

```diff
@@ -19,6 +19,7 @@ public static partial class HttpClientBuilderExtensions
         public static Microsoft.Extensions.DependencyInjection.IHttpClientBuilder ConfigurePrimaryHttpMessageHandler(this Microsoft.Extensions.DependencyInjection.IHttpClientBuilder builder, System.Func<System.Net.Http.HttpMessageHandler> configureHandler) { throw null; }
         public static Microsoft.Extensions.DependencyInjection.IHttpClientBuilder ConfigurePrimaryHttpMessageHandler<THandler>(this Microsoft.Extensions.DependencyInjection.IHttpClientBuilder builder) where THandler : System.Net.Http.HttpMessageHandler { throw null; }
         public static Microsoft.Extensions.DependencyInjection.IHttpClientBuilder SetHandlerLifetime(this Microsoft.Extensions.DependencyInjection.IHttpClientBuilder builder, System.TimeSpan handlerLifetime) { throw null; }
+        public static Microsoft.Extensions.DependencyInjection.IHttpClientBuilder SetLogSensitiveHeaders(this Microsoft.Extensions.DependencyInjection.IHttpClientBuilder builder, System.Collections.Generic.ICollection<string> logSensitiveHeaders) { throw null; }
```

Comment:
I think this is overall looking pretty good. 

If we keep the current design approach. I have some suggestions about naming/terminology that I think will make things clearer.

```
public static Microsoft.Extensions.DependencyInjection.IHttpClientBuilder RedactLoggedHeaderValues(this Microsoft.Extensions.DependencyInjection.IHttpClientBuilder builder, System.Collections.Generic.ICollection<string> logSensitiveHeaders)

public System.Collections.Generic.IList<string> RedactedLoggedHeaderNames { [System.Runtime.CompilerServices.CompilerGeneratedAttribute] get { throw null; } }
```

I like using the term *redact* here - because it's exactly what this does. It still logs whether or not the header was present, but doesn't log the value.

----

Another design idea that comes up here is to make this a predicate-based system. The usage would look like:

```
services.AddHttpClient("Foo").RedactLoggedHeaders((header) =>
{
    if (header == "Authorize")
    {
        return false;
    }

    return true;
});
```

This is nice because the app developer can write any code they want, and we don't need to add any additional features. We could keep the functionality you added in this PR (deny-list) and just layer it on top as an overload.

```
services.AddHttpClient("Foo").RedactLoggedHeaders(new []{ "Authorize", });
```

If we wanted to pivot this to support an allow-list, you could have that too, without really changing the logic.

```
services.AddHttpClient("Foo").AllowLoggedHeaders(new []{ "Content-Type", "Content-Length", });
```